### PR TITLE
feat: Walk API classes

### PR DESCRIPTION
**Pull request by seanpearsonuk**
Add a tool to walk static API classes, already being used for analysing the API through separate code.

### CODE REVIEW

**Comment by mkundu1** [comment]
@seanpearsonuk For NamedObject and ListObject, we have to get the `child_object_type` property which is a Group and then traverse inside that Group class.

File: src/ansys/fluent/core/codegen/walk_api.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    # Skip the root path
+    if current_path:
+        on_each_path(current_path)
+
+    # Get child names and their respective classes
+    child_names = getattr(api_root_cls, "child_names", [])
+    child_classes = getattr(api_root_cls, "_child_classes", {})
+
+    # Traverse each child
+    for child_name in child_names:
```
Comment (mkundu1) [inline src/ansys/fluent/core/codegen/walk_api.py:37]: We also need to loop over `command_names` and `query_names`

Comment (mkundu1) [reply src/ansys/fluent/core/codegen/walk_api.py:37]: and `argument_names` for Command and Query classes

Comment (seanpearsonuk) [reply src/ansys/fluent/core/codegen/walk_api.py:37]: In a later version, I also want to pass each item's base type info to the callback for type-specific checks.

**Review by mkundu1 — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

**Pull request merged**